app/src/main/java/org/oppia/android/app/mydownloads/MyDownloadsActivityPresenter.kt

```diff
@@ -1,28 +1,75 @@
 package org.oppia.android.app.mydownloads
 
+import android.view.View
 import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.Toolbar
+import androidx.databinding.DataBindingUtil
+import androidx.drawerlayout.widget.DrawerLayout
+import androidx.viewpager2.widget.ViewPager2
+import com.google.android.material.tabs.TabLayout
+import com.google.android.material.tabs.TabLayoutMediator
 import org.oppia.android.R
 import org.oppia.android.app.activity.ActivityScope
+import org.oppia.android.app.drawer.NavigationDrawerFragment
+import org.oppia.android.databinding.MyDownloadsActivityBinding
 import javax.inject.Inject
 
+const val INTERNAL_PROFILE_ID_SAVED_KEY = "MyDownloadsActivity.internal_profile_id"
+const val IS_ALLOWED_DOWNLOAD_ACCESS_SAVED_KEY = "MyDownloadsActivity.is_allowed_download_access"
+
 /** The presenter for [MyDownloadsActivity]. */
 @ActivityScope
 class MyDownloadsActivityPresenter @Inject constructor(private val activity: AppCompatActivity) {
-  fun handleOnCreate() {
-    activity.setContentView(R.layout.my_downloads_activity)
-    if (getMyDownloadsFragment() == null) {
-      activity.supportFragmentManager.beginTransaction().add(
-        R.id.my_downloads_fragment_placeholder,
-        MyDownloadsFragment()
-      ).commitNow()
+
+  private lateinit var navigationDrawerFragment: NavigationDrawerFragment
+  private lateinit var toolbar: Toolbar
+  private var internalProfileId: Int = -1
+
+  fun handleOnCreate(internalProfileId: Int, isFromNavigationDrawer: Boolean) {
+    val binding =
+      DataBindingUtil.setContentView<MyDownloadsActivityBinding>(
+        activity,
+        R.layout.my_downloads_activity
+      )
+    binding.apply {
+      lifecycleOwner = activity
+    }
+    toolbar = binding.myDownloadsActivityToolbar
+    activity.setSupportActionBar(toolbar)
+    activity.supportActionBar!!.title = activity.getString(R.string.menu_my_downloads)
```

Comment:
There are already two strings for "My Downloads" in `strings.xml` file and I have suggested one more in manifest file.

Suggest keeping only two strings
`menu_my_downloads` and `my_downloads_activity_title ` and use it correctly. For example in this file `menu_my_downloads` should not be used.

Reply:
done